lib/commands/web.js

```diff
@@ -0,0 +1,193 @@
+    throw new errors.NotImplementedError();
+  }
+};
+
+commands.getUrl = async function () {
+  if (this.isWebContext()) {
+    this.processingRemoteCmd = true;
+    let value = await this.remote.execute('window.location.href');
+    this.processingRemoteCmd = false;
+    return value;
+  } else {
+    throw new errors.NotImplementedError();
+  }
+};
+
+commands.title = async function () {
+  if (this.isWebContext()) {
+    let res =  await this.executeAtom('title', []);
+    return res.value;
+  } else {
+    throw new errors.NotImplementedError();
+  }
+};
+
+commands.findWebElementOrElements = async function (strategy, selector, ctx, many) {
+  let atomsElement = this.getAtomsElement(ctx);
+  let res;
+  let doFind = async () => {
+    res = await this.executeAtom(`find_element${many ? 's' : ''}`, [strategy, selector, atomsElement]);
+    return res.status === 0 && res.value !== null;
+  };
+  await this.implicitWaitForCondition(doFind);
+  return res.value;
+};
+
+commands.click = async function (elementId) {
+  if (this.isWebContext()) {
+    let atomsElement = this.useAtomsElement(elementId);
+    return await this.executeAtom('click', [atomsElement]);
+  } else {
+    if (this.useRobot) {
+      // let locCmd = `au.getElement('${elementId}').rect()`;
+      // let res = await this.uiAutoClient.sendCommand(locCmd);
+      // let rect = res.value;
+      // let pos = {x: rect.origin.x, y: rect.origin.y};
+      // let size = {w: rect.size.width, h: rect.size.height};
+      // let tapPoint = { x: pos.x + (size.w / 2), y: pos.y + (size.h / 2) };
+      // let tapUrl = `${this.args.robotUrl}/tap`;
+      // request.post({url:tapUrl, form: {x:tapPoint.x, y:tapPoint.y}}, cb);
+      /* TODO */throw new errors.NotYetImplementedError();
+    } else {
+      await this.nativeTap(elementId);
+    }
+  }
+};
+
+helpers.executeAtom = async function (atom, args, alwaysDefaultFrame = true) {
+  let counter = this.executedAtomsCounter++;
+  let frames = alwaysDefaultFrame === true ? [] : this.curWebFrames;
+  this.returnedFromExecuteAtom[counter] = false;
+  this.processingRemoteCmd = true;
+  let res = await this.remote.executeAtom(atom, args, frames);
+  this.processingRemoteCmd = false;
+  if (!this.returnedFromExecuteAtom[counter]) {
+    this.returnedFromExecuteAtom[counter] = true;
+    res = this.parseExecuteResponse(res);
+  } else {
+    // try to get rid of any alert 10 times, with 5s interval
+    await retryInterval(10, 5000, this.lookForAlert.bind(this), counter);
+  }
+
+  return res;
+};
+
+helpers.lookForAlert = async function (counter) {
+  if (!_.isNull(this.instruments) && !_.isNull(this.uiAutoClient)) {
+    if (!this.returnedFromExecuteAtom[counter] && !this.selectingNewPage) {
+      logger.debug('atom did not return yet, checking to see if ' +
+                   'we are blocked by an alert');
+      // temporarily act like we're not processing a remote command
```

Comment:
is this comment still relevant? i don't see us setting any processingRemoteCommand variable here